### ISSUE TITLE
As a Content Editor I want to link a source to a document as a footnote, in order to show that the source is helpful for understanding the document.

### ISSUE THREAD

**Issue by thatbudakguy** [Princeton-CDH/geniza#109]
**Is your feature request related to a problem? Please describe.**
This depends on #96 and represents a relationship between a document and the scholarship model created in that issue.

## Testing Notes
- [x] Go to a document and create a footnote on that document (Make sure that you've already created a source). Click save ensure that footnote is attached to the document.
- [x] Create at least two footnotes with different document relationships and one footnote with more than one document relationship selected
- [x] Confirm that the footnote display on the document looks appropriate.
- [ ] Go to the Footnote admin list view. Confirm that the list page and the change page looks appropriate.
- [x] Test that you can filter correctly on document relation, including footnotes with more than one

## Dev Notes

- [x] Create a footnote model
   * Fields: Generic Foreign Key, ForeignKey(Source), Document Relation field, Notes, page range
- [x] Register footnote model in admin
- [x] Include footnote in document model as generic foreign key
- [x] Place inline footnote in document admin change view
- [x] Update permissions to include footnote model

**Comment by mrustow** [comment]
I tried to link an existing source to a document. I searched for the source by author and it came up: yay! I chose it and added page numbers. But then, I hit save, and got an error message! Egads!
<img width="952" alt="Screen Shot 2021-04-09 at 4 47 58 PM" src="https://user-images.githubusercontent.com/73319225/114239022-82c1b400-9953-11eb-8f70-f3b7f21efd93.png">
<img width="953" alt="Screen Shot 2021-04-09 at 4 48 09 PM" src="https://user-images.githubusercontent.com/73319225/114239023-835a4a80-9953-11eb-8715-158a7e7c08c4.png">

**Comment by rlskoeser** [comment]
I can't tell from the screenshot what the error is — what field is it complaining about? Do we have something set to required that shouldn't be?

**Comment by mrustow** [comment]
I can't tell either. It's not pointing to a field, just complaining, kind of like our kids.

**Comment by richmanrachel** [comment]
I was able to add a footnote as Content Admin, but Marina is still getting errors as Content Editor.

**Comment by mrustow** [comment]
I tried it as content editor. I also tried it with and without footnotes. Then I tried adding a second source. No dice.

**Comment by richmanrachel** [comment]
Feature idea: being able to see all the documents a source is related to from the source page.

**Comment by rlskoeser** [comment]
Oh, OK!  If content admin can do it but content editor can't we missed something with the permissions. That's probably enough information for us to figure it out.

**Comment by rlskoeser** [comment]
@kmcelwee this failed testing because the content editor permissions aren't set correctly to allow adding footnotes. Can you investigate and see what permission we missed? Maybe content editors need to be able to delete footnotes in order to use the inline admin editor.

**Comment by kmcelwee** [comment]
@thatbudakguy and I worked through this, and were unable to replicate a permissions error. We did receive a similar error when we didn't make a selection in the required "document relation type" checkbox. We can't see it in the provided screenshot. @mrustow would you mind double checking that that field is selected? It's easy to miss.

**Comment by richmanrachel** [comment]
> Go to the Footnote admin list view. Confirm that the list page and the change page looks appropriate.
- While you can see in the list view which shelfmarks the footnote is connected to, for several real sources there will be hundreds of entries, so this section will become unwieldy. 
- We also can't see this list from inside the footnotes page yet, and it would be much more efficient to add batches of footnotes from the footnote and/or source page than to add it to each document individually.

**Comment by richmanrachel** [comment]
And I just tested adding a footnote from the Content Editor (majhul) account and it worked!

@rlskoeser - Should I close this and create a new issue for how/where to see footnote/source connections?

**Comment by rlskoeser** [comment]
@richmanrachel yes, let's do that — I think it will be easier to think about and discuss when we have a bunch of test data imported into the system (pending #108 ). We didn't add very many filters on the footnote admin, but we may want them once there we have more content.

I wasn't aware there would be a need for adding batches of footnotes — we can discuss workflows that might be more efficient with the current system and/or improvements to make it more efficient.

**Comment by richmanrachel** [comment]
Apologies for leaving this on hold! Closing and writing new user story now.
